pydatastructs/miscellaneous_data_structures/stack.py

```diff
@@ -53,18 +54,25 @@ class Stack(object):
     """
 
     def __new__(cls, implementation='array', **kwargs):
-        raise_if_backend_is_not_python(
-            cls, kwargs.get('backend', Backend.PYTHON))
+        backend = kwargs.get('backend', Backend.PYTHON)
         if implementation == 'array':
-            return ArrayStack(
-                kwargs.get('items', None),
-                kwargs.get('dtype', int))
+            items = kwargs.get('items', None)
+            dtype = kwargs.get('dtype', int)
+            if backend == Backend.CPP:
+                if items is None:
+                    return _stack.ArrayStack(dtype)
+                else:
+                    return _stack.ArrayStack(items, dtype)
+            else:
+                return ArrayStack(items, dtype)
```

Comment:
It would be great if we can handle this in C++ backend itself? Specifically I would just do `_stack.ArrayStack(**kwargs)` and then C++ backend should try to fetch `item`, `dtype` from `**kwargs` using `PyObject_GetItem` (confirm from Python C-API docs).

Can you try it out?

Reply:
Yep, we can send positional args and named args now